pkg/knuu/knuu.go

```diff
@@ -54,10 +56,72 @@ func InitializeWithIdentifier(uniqueIdentifier string) error {
 		return err
 	}
 
+	// read timeout from env
+	timeoutString := os.Getenv("KNUU_TIMEOUT")
+
+	if timeoutString == "" {
+		timeout = 60 * time.Minute
+	} else {
+		parsedTimeout, err := time.ParseDuration(timeoutString)
+		if err != nil {
+			return fmt.Errorf("cannot parse timeout: %s", err)
+		}
+		timeout = parsedTimeout
+	}
+
+	if err := handleTimeout(); err != nil {
+		return fmt.Errorf("cannot handle timeout: %s", err)
+	}
+
 	return nil
 }
 
 // IsInitialized returns true if knuu is initialized, and false otherwise
 func IsInitialized() bool {
 	return k8s.IsInitialized()
 }
+
+// handleTimeout creates a timeout handler that will delete all resources with the identifier after the timeout
+func handleTimeout() error {
+
+	instance, err := NewInstance("timeout-handler")
+	if err != nil {
+		return fmt.Errorf("cannot create instance: %s", err)
+	}
+	// FIXME: use supported kubernetes version images (use of latest could break)
+	if err := instance.SetImage("docker.io/bitnami/kubectl:latest"); err != nil {
+		return fmt.Errorf("cannot set image: %s", err)
+	}
+	if err := instance.Commit(); err != nil {
+		return fmt.Errorf("cannot commit instance: %s", err)
+	}
+	timeoutSeconds := int64(timeout.Seconds())
+
+	// command to wait for timeout and delete all resources with the identifier
+	var command = []string{"sh", "-c"}
+	cmd := fmt.Sprintf("sleep %d && kubectl delete all,pvc,netpol,roles,serviceaccounts,rolebindings -l test-run-id=%s -n %s --wait=false", timeoutSeconds, identifier, k8s.Namespace())
```

Comment:
[non-blocking]: Isn't it better to use the go client api for this? Just curious why you opted for this

Reply:
This command runs in a container in the cluster. Not on the machine where the test is executed. To make it as minimal and easy as possible, I just run a simple image like a Pod and run a shell. The alternative would be to create a separate Go application. I was considering it but decided to go for the simple solution.

Reply:
then this approach sounds sensible to me. Woud you mind putting this as a comment in the cmd var? 🙏